astropy/nddata/nddata.py

```diff
@@ -180,10 +178,25 @@ def __repr__(self):
 
     @property
     def data(self):
+        """
+        `~numpy.ndarray`: the data
+
+        Even though it is only enforced that the `data` is `~numpy.ndarray`
+        which does not contain objects. For use with `NDArithmeticMixin` it is
+        assumed that the data array only contains numbers.
```

Comment:
I'd move the the sentence that starts `For use with` to the `NDArithmeticMixin`.